src/backend/distributed/executor/subplan_execution.c

```diff
@@ -19,6 +19,10 @@
 #include "executor/executor.h"
 
 
+int MaxIntermediateResult = 1048576; /* maximum size in KB the intermediate result can grow to */
+bool UseResultSizeLimit = false;
```

Comment:
Maybe add a comment that when this is true, we enforce intermediate result size limit in all executors.